LogSearch_Test.LogSearch randomly fails

I'm running unittest on Ubuntu 20.04 in a Docker container. It fails with a probability of about 50%.

```
$ ./dbms/gtests_dbms --gtest_filter='LogSearch_Test.LogSearch'
[2021/07/09 15:25:26.170 +08:00] [INFO] [<unknown>] ["PathCapacityMetrics: Init capacity [path=/pingcap/tics/build/tmp/] [capacity=0.00 B]"] [thread_id=1]
[2021/07/09 15:25:26.171 +08:00] [DEBUG] [<unknown>] ["PageStorage::getMaxDataVersion: getMaxDataVersion done from MergingReader of PageFile_4_0, type: Formal, sequence no: 0, meta offset: 0, data offset: 0 [max version=2]"] [thread_id=1]
[2021/07/09 15:25:26.171 +08:00] [INFO] [<unknown>] ["RegionPersister: RegionPersister running in normal mode"] [thread_id=1]
[2021/07/09 15:25:26.172 +08:00] [INFO] [<unknown>] ["PageStorage: RegionPersister begin to restore data from disk. [path=/pingcap/tics/build/tmp/kvstore] [num_writers=4]"] [thread_id=1]
[2021/07/09 15:25:26.172 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister merge done from MergingReader of PageFile_1_0, type: Formal, sequence no: 0, meta offset: 0, data offset: 0"] [thread_id=1]
[2021/07/09 15:25:26.172 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister merge done from MergingReader of PageFile_2_0, type: Formal, sequence no: 0, meta offset: 0, data offset: 0"] [thread_id=1]
[2021/07/09 15:25:26.172 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister merge done from MergingReader of PageFile_3_0, type: Formal, sequence no: 0, meta offset: 0, data offset: 0"] [thread_id=1]
[2021/07/09 15:25:26.172 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister merge done from MergingReader of PageFile_4_0, type: Formal, sequence no: 0, meta offset: 0, data offset: 0"] [thread_id=1]
[2021/07/09 15:25:26.173 +08:00] [INFO] [<unknown>] ["PageStorage: RegionPersister restore 0 pages, write batch sequence: 0, 0 puts and 0 refs and 0 deletes and 0 upserts"] [thread_id=1]
[2021/07/09 15:25:26.173 +08:00] [INFO] [<unknown>] ["KVStore: Restored 0 regions. "] [thread_id=1]
[2021/07/09 15:25:26.173 +08:00] [INFO] [<unknown>] ["RegionTable: Start to restore"] [thread_id=1]
[2021/07/09 15:25:26.173 +08:00] [INFO] [<unknown>] ["RegionTable: Restore 0 tables"] [thread_id=1]
[2021/07/09 15:25:26.173 +08:00] [INFO] [<unknown>] ["BackgroundProcessingPool: Create BackgroundProcessingPool with 16 threads"] [thread_id=1]
[2021/07/09 15:25:26.176 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister Before gc, 0 puts and 0 refs and 0 deletes and 0 upserts"] [thread_id=2]
[2021/07/09 15:25:26.176 +08:00] [DEBUG] [<unknown>] ["PageStorage: RegionPersister gcApply remove 2 invalid snapshots, 1 snapshots left, longest lifetime 0.000 seconds, created from thread_id 0"] [thread_id=2]
[2021/07/09 15:25:26.176 +08:00] [TRACE] [<unknown>] ["PageStorage: RegionPersister GC exit with no files to gc."] [thread_id=2]
Note: Google Test filter = LogSearch_Test.LogSearch
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from LogSearch_Test
[ RUN      ] LogSearch_Test.LogSearch
[2021/07/09 15:25:26.183 +08:00] [ERROR] [<unknown>] ["LogIterator: readLog error: level: 6/"] [thread_id=1]
/pingcap/tics/dbms/src/Flash/tests/gtest_log_search.cpp:23: Failure
Value of: log.has_value()
  Actual: false
Expected: true
[  FAILED  ] LogSearch_Test.LogSearch (1 ms)
[----------] 1 test from LogSearch_Test (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] LogSearch_Test.LogSearch

 1 FAILED TEST
```

The error message also prints random strings after `readLog error: level: `. e.g.

```
[2021/07/09 15:30:39.099 +08:00] [ERROR] [<unknown>] ["LogIterator: readLog error: level: 9�u0006"] [thread_id=1]
[2021/07/09 15:32:02.476 +08:00] [ERROR] [<unknown>] ["LogIterator: readLog error: level: ��"] [thread_id=1]
[2021/07/09 15:32:12.206 +08:00] [ERROR] [<unknown>] ["LogIterator: readLog error: level: pf"] [thread_id=1]
```

--- a/dbms/src/Flash/LogSearch.cpp
+++ b/dbms/src/Flash/LogSearch.cpp
@@ -109,7 +109,7 @@ LogIterator::Result<LogIterator::LogEntry> LogIterator::readLog()
         &timezone_hour, &timezone_min, level_buff);
 
     {
-        std::tm time;
+        std::tm time {};
         time.tm_year = year - 1900;
         time.tm_mon = month - 1;
         time.tm_mday = day;

